src/Simplify_And.cpp

```diff
@@ -57,6 +57,7 @@ Expr Simplify::visit(const And *op, ExprInfo *bounds) {
          rewrite(!x && x, false) ||
          rewrite(y <= x && x < y, false) ||
          rewrite(x != c0 && x == c1, b, c0 != c1) ||
+         rewrite(x == c0 && x == c1, false, c0 != c1) ||
```

Comment:
Is this strictly *necessary* or just a drive-by optimization? (Wondering if this could be landed separately to try to flush out the weird failure)